sdk/rust/oak/src/lib.rs

```diff
@@ -298,11 +299,19 @@ pub fn channel_write(half: WriteHandle, buf: &[u8], handles: &[Handle]) -> Resul
     result_from_status(status as i32, ())
 }
 
-/// Create a new unidirectional channel.
+/// Create a new unidirectional channel with the specified label.
+///
+/// Only nodes with a compatible label will be able to read and write to this channel:
+///
+/// - A node with label `L_w` may write to a channel with label `L_c` iff `L_w ⊑ L_c`.
+/// - A node with label `L_r` may read from a channel with label `L_c` iff `L_c ⊑ L_r`.
```

Comment:
I do not see how allowing a node to change its own label works together with the idea of labels being public. If a node can update a publicly visible value, it can be used as a channel to leak information. Even though it can only ever increase its own label, a specifically chose sequence of increases can be used to encode information.